addons/mail/static/src/models/message_composition/message_composition.js

```diff
@@ -0,0 +1,201 @@
+/** @odoo-module **/
+
+import { emojis } from '@mail/js/emojis';
+import { registerModel } from '@mail/model/model_core';
+import { unlink, insertAndReplace, replace } from '@mail/model/model_field_command';
+import { attr, many, one } from '@mail/model/model_field';
+import { addLink, escapeAndCompactTextContent, parseAndTransform } from '@mail/js/utils';
+
+import { escape } from '@web/core/utils/strings';
+
+/**
+ * Model of a message being composed in a composer.
+ *
+ * Note that a message composition is not a full-fledged message, because it
+ * hasn't been posted yet.
+ * Still, it looks somewhat like a message, which is intended because when a
+ * message composition is posted, it becomes a message.
+ */
+registerModel({
+    name: 'MessageComposition',
+    identifyingFields: ['id'],
```

Comment:
Not a good choice for identifying field.
The concept of `MessageComposition/id` is poor: it's determined by flow, so imperative design, which goes against desired declarative design.

A `MessageComposition` is a message being composed in a certain context. I would expect its identify being on something close to `thread` or `threadView` or `messageView`. Maybe not the views, but certainly `thread`.

Reply:
I don't understand this point, what is the purpose of using thread (non unique value) as identifying fields (should be somehow unique?)?
This model doesn't even need identifying fields to be honest, since it's never identified...

Reply:
All records should be identified. It's essential as it gives the whole reason for this record to exist. It gives reason to have this entity: it needs to have an identity (in the future, could have several, but need at least one to exist).

I'm not saying the identifying fields of `MessageComposition` should simply be `thread`.
I just gave some ideas of fields that can be in the identifying fields, like `thread`.

TBH, I'm not entirely sure how a message composition should be uniquely identified.
Just that `MessageComposition/id` is not a good concept.

In terms of UI, it's easy to see `MessageComposition` being very close to `ComposerView`. But it's not exactly a view we want as identifying field, because we want a `MessageComposition` to exist even when the views don't exist.
So it's somehow the logic part of `ComposerView` that is not coupled to the UI... Yeah... It's a granularity of model-making that we don't really have as of now... A sort of record that has state of view even when it's no longer in the UI.

Right now we made shortcut of defining such a record as the shared state of the views. So in your case, this is `Composer`. `Composer` is more-or-less arbitrary split of `Thread`, hence why I was thinking of `thread` as the (or a part) of identifying field(s).

Now that I think about it... `MessageComposition/thread` should actually be `MessageComposition/composer`. It's no big deal now, but it's important for the future when we want to break the rough shortcut we have of "Thread" = "Composer".

Reply:
It doesn't have to be an incremental id, or be called `id`, but it fundamentally needs a unique id. There is nothing else to identify it than the user action that generated it. We could use `ev` if we really wanted, but that will not serialize well to string I don't think (must be due to current implementation of identifying).

It's fine to link it also to thread/composer if we really want, but as soon as there is a unique id in the mix, any other identifying field become unnecessary.